django/test/utils.py

```diff
@@ -196,6 +189,17 @@ def setup_databases(verbosity, interactive, keepdb=False, debug_sql=False, paral
     return old_names
 
 
+def clone_databases(verbosity, keepdb, parallel):
+    for alias in connections:
```

Comment:
I ran into an interesting bug with this. If you define two databases:

```
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.mysql',
        ....
    },
    'other': {
        'ENGINE': 'django.db.backends.mysql',
        ....
    }
}
```

And execute the `model_options` test suite, `mysqldump` will be executed with `''` as the database name for the `other` databases. I tracked this down to the call to `databases = self.get_databases(suite)` in `run_tests()` - this returns a _subset_ of databases that the test suite requires to run. In the case of `model_options` it only requires the `default` database.

Only these selected databases have a `NAME` set to `test_`, which means `connections['other'].settings_dict['NAME']` is an empty string.

Long story short - I think you need to pass in the `databases` set [from this line here](https://github.com/django/django/blob/297df97119639c0e49b9a26fce60ce2ff41e0e73/django/test/runner.py#L743) into this method.

Reply:
Ah yes, the databases created depend on which databases are needed by the test suite. Good catch!